packages/browser/src/index.ts

```diff
@@ -1,377 +1,76 @@
-    const accessToken = this.accessTokenMap.get(accessTokenKey);
-
-    if (accessToken && accessToken.expiresAt > Date.now() / 1000) {
-      return accessToken.token;
-    }
-
-    // Since the access token has expired, delete it from the map.
-    if (accessToken) {
-      this.accessTokenMap.delete(accessTokenKey);
-    }
-
-    /**
-     * Need to fetch a new access token using refresh token.
-     * Reuse the cached promise if exists.
-     */
-    const cachedPromise = this.getAccessTokenPromiseMap.get(accessTokenKey);
+  return response.json<T>();
+};
 
-    if (cachedPromise) {
-      return cachedPromise;
-    }
+const handleRedirect = (url: string) => {
+  window.location.assign(url);
+};
 
-    /**
-     * Create a new promise and cache in map to avoid race condition.
-     * Since we enable "refresh token rotation" by default,
-     * it will be problematic when calling multiple `getAccessToken()` closely.
-     */
-    const promise = this.getAccessTokenByRefreshToken(resource);
-    this.getAccessTokenPromiseMap.set(accessTokenKey, promise);
+const logtoStorageItemKeyPrefix = `logto`;
 
-    const token = await promise;
-    this.getAccessTokenPromiseMap.delete(accessTokenKey);
+class BrowserStorage implements Storage {
+  private readonly storageKey: string;
 
-    return token;
+  constructor(appId: string) {
+    this.storageKey = `${logtoStorageItemKeyPrefix}:${appId}`;
   }
 
-  public getIdTokenClaims(): IdTokenClaims {
-    if (!this.idToken) {
-      throw new LogtoClientError('not_authenticated');
+  getItem(key: 'idToken' | 'refreshToken' | 'accessToken' | 'signInSession'): Nullable<string> {
+    if (key === 'signInSession') {
+      return sessionStorage.getItem(this.storageKey);
     }
 
-    return decodeIdToken(this.idToken);
-  }
-
-  public async signIn(redirectUri: string) {
-    const { appId: clientId, prompt, resources, scopes } = this.logtoConfig;
-    const { authorizationEndpoint } = await this.getOidcConfig();
-    const codeVerifier = generateCodeVerifier();
-    const codeChallenge = await generateCodeChallenge(codeVerifier);
-    const state = generateState();
-
-    const signInUri = generateSignInUri({
-      authorizationEndpoint,
-      clientId,
-      redirectUri,
-      codeChallenge,
-      state,
-      scopes,
-      resources,
-      prompt,
-    });
-
-    this.signInSession = { redirectUri, codeVerifier, state };
-    this.refreshToken = null;
-    this.idToken = null;
-
-    window.location.assign(signInUri);
+    return localStorage.getItem(`${this.storageKey}:${key}`);
   }
 
-  public isSignInRedirected(url: string): boolean {
-    const { signInSession } = this;
-
-    if (!signInSession) {
-      return false;
+  setItem(key: 'idToken' | 'refreshToken' | 'accessToken' | 'signInSession', value: string): void {
+    if (key === 'signInSession') {
+      sessionStorage.setItem(this.storageKey, value);
+    } else {
+      localStorage.setItem(`${this.storageKey}:${key}`, value);
```

Comment:
better to use return early or switch with deafult?

Reply:
updated